lib/executor/variable_looping_vus_test.go

```diff
@@ -1002,3 +1004,115 @@ func TestSegmentedIndex(t *testing.T) {
 		assert.EqualValues(t, 0, s.scaled)
 	})
 }
+
+// TODO: delete in favor of lib.generateRandomSequence() after
+// https://github.com/loadimpact/k6/issues/1302 is done (can't import now due to
+// import loops...)
+func generateRandomSequence(t testing.TB, n, m int64, r *rand.Rand) lib.ExecutionSegmentSequence {
+	var err error
+	ess := lib.ExecutionSegmentSequence(make([]*lib.ExecutionSegment, n))
+	numerators := make([]int64, n)
+	var denominator int64
+	for i := int64(0); i < n; i++ {
+		numerators[i] = 1 + r.Int63n(m)
+		denominator += numerators[i]
+	}
+	from := big.NewRat(0, 1)
+	for i := int64(0); i < n; i++ {
+		to := new(big.Rat).Add(big.NewRat(numerators[i], denominator), from)
+		ess[i], err = lib.NewExecutionSegment(from, to)
+		require.NoError(t, err)
+		from = to
+	}
+
+	return ess
+}
+
+func TestSumRandomSegmentSequenceMatchesNoSegment(t *testing.T) {
+	t.Parallel()
+
+	seed := time.Now().UnixNano()
+	r := rand.New(rand.NewSource(seed))
+	t.Logf("Random source seeded with %d\n", seed)
+
+	const (
+		numTests         = 10
+		maxStages        = 10
+		minStageDuration = 1 * time.Second
+		maxStageDuration = 10 * time.Minute
+		maxVUs           = 300
+		segmentSeqMaxLen = 15
+		maxNumerator     = 300
+	)
+	getTestConfig := func(name string) VariableLoopingVUsConfig {
+		stagesCount := 1 + r.Int31n(maxStages)
+		stages := make([]Stage, stagesCount)
+		for s := int32(0); s < stagesCount; s++ {
+			dur := time.Duration(r.Int63n(int64(maxStageDuration - minStageDuration))).Round(time.Second)
+			stages[s] = Stage{Duration: types.NullDurationFrom(dur), Target: null.IntFrom(r.Int63n(maxVUs))}
+		}
+
+		c := NewVariableLoopingVUsConfig(name)
+		c.GracefulRampDown = types.NullDurationFrom(0)
+		c.GracefulStop = types.NullDurationFrom(0)
+		c.StartVUs = null.IntFrom(r.Int63n(maxVUs))
+		c.Stages = stages
+		return c
+	}
+
+	subtractChildSteps := func(t *testing.T, parent, child []lib.ExecutionStep) {
+		t.Logf("subtractChildSteps()")
+		for i := 0; i < len(child); i++ {
+			t.Logf("	child planned VUs for time offset %s: %d", child[i].TimeOffset, child[i].PlannedVUs)
+		}
```

Comment:
this would've been more readable as 
```suggestion
		for _, step := range child {
			t.Logf("	child planned VUs for time offset %s: %d", step.TimeOffset, step.PlannedVUs)
		}
```

Reply:
Resolved by https://github.com/loadimpact/k6/pull/1413/commits/d47a00bec8d8ccbe4f616982c78b496ce78da136